tests/test_unit/test_kinematics.py

```diff
@@ -182,3 +184,120 @@ def test_approximate_derivative_with_invalid_order(order):
     expected_exception = ValueError if isinstance(order, int) else TypeError
     with pytest.raises(expected_exception):
         kinematics._compute_approximate_time_derivative(data, order=order)
+
+
+class TestNavigation:
+    """Test suite for navigation-related functions in the kinematics module."""
+
+    @pytest.fixture
+    def mock_dataarray(self):
+        """Return a mock DataArray containing four known head orientations."""
+        time = [0, 1, 2, 3]
+        individuals = ["individual_0"]
+        keypoints = ["left_ear", "right_ear", "nose"]
+        space = ["x", "y"]
+
+        ds = xr.DataArray(
+            [
+                [[[1, 0], [-1, 0], [0, -1]]],  # time 0
+                [[[0, 1], [0, -1], [1, 0]]],  # time 1
+                [[[-1, 0], [1, 0], [0, 1]]],  # time 2
+                [[[0, -1], [0, 1], [-1, 0]]],  # time 3
+            ],
+            dims=["time", "individuals", "keypoints", "space"],
+            coords={
+                "time": time,
+                "individuals": individuals,
+                "keypoints": keypoints,
+                "space": space,
+            },
+        )
+        return ds
+
+    @pytest.fixture
+    def mock_dataarray_3D(self):
+        """Return a 3D DataArray containing a known head orientation."""
+        time = [0]
+        individuals = ["individual_0"]
+        keypoints = ["left", "right"]
+        space = ["x", "y", "z"]
+
+        ds = xr.DataArray(
+            [
+                [[[1, 0, 0], [-1, 0, 0]]],
+            ],
+            dims=["time", "individuals", "keypoints", "space"],
+            coords={
+                "time": time,
+                "individuals": individuals,
+                "keypoints": keypoints,
+                "space": space,
+            },
+        )
+        return ds
+
+    def test_compute_head_direction_vector(
+        self, mock_dataarray, mock_dataarray_3D
+    ):
+        """Test that the correct head direction vectors
+        are computed from a basic mock dataset.
+        """
+        # Test that validators work
+
+        # Catch incorrect datatype
```

Comment:
I would suggest splitting this test in three, as the behaviour is now a bit conflated.
- One test to check `compute_head_vector` with invalid inputs.
- another test to check it with valid inputs, 
- and another test to check its behaviour with nan values.

I would make different fixtures for each of them. I think the test for invalid inputs could be easily parametrised (it is a bit easier to debug and read that way). 

This is the structure followed for the rest of the kinematic tests - see for example [here](https://github.com/neuroinformatics-unit/movement/blob/f4f46dfc48dd1dde5959b01fb0a17fad93070ae9/tests/test_unit/test_kinematics.py#L166) (but in this case it would be a bit easier cause we don't parametrise over the kinematic function).